File: Greedy/0605.Can Place Flowers/Solution.java
class Solution {
    public boolean canPlaceFlowers(int[] flowerbed, int n) {
        int m = flowerbed.length;
        int cnt = 0;
        for (int i = 0; i < m; i++) {
            if (flowerbed[i] != 0) {
                continue;
            }
            if (flowerbed[Math.max(0, i - 1)] == 0 && flowerbed[Math.min(m - 1, i + 1)] == 0) {
                flowerbed[i] = 1;
                cnt++;
            }
        }
        return cnt >= n;
    }
}

File: Greedy/0605.Can Place Flowers/Solution.py
class Solution:
    def canPlaceFlowers(self, flowerbed: List[int], n: int) -> bool:
        m = len(flowerbed)
        cnt = 0
        for i in range(m):
            if flowerbed[i]:
                continue
            if not flowerbed[max(i - 1, 0)] and not flowerbed[min(m - 1, i + 1)]:
                flowerbed[i] = 1
                cnt += 1
        return cnt >= n